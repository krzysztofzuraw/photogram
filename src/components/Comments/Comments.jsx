import React from "react";
import { connect } from "react-redux";

import { Comment, Header, Form, Button } from "semantic-ui-react";
import { addComment } from "../../actions/index.js";

class Comments extends React.Component {
  constructor(props) {
    super(props);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(event) {
    event.preventDefault();
    this.props.addComment(this.refs.author.value, this.refs.comment.value);
    this.refs.author.value = "";
    this.refs.comment.value = "";
  }

  render() {
    return (
      <Comment.Group>
        {this.props.comments.map((comment, key) => (
          <Comment key={key}>
            <Comment.Content>
              <Comment.Author>{comment.author}</Comment.Author>
              <Comment.Text>{comment.text}</Comment.Text>
            </Comment.Content>
          </Comment>
        ))}

        <Form onSubmit={this.handleSubmit} reply>
          <Form.Field>
            <input placeholder="Name" type="text" required ref="author" />
          </Form.Field>
          <Form.Field>
            <input placeholder="Comment" type="text" ref="comment" />
          </Form.Field>
          <Button
            content="Add Reply"
            labelPosition="left"
            icon="edit"
            primary
          />
        </Form>
      </Comment.Group>
    );
  }
}

const mapStateToProps = state => ({
  comments: state.comments
});

const mapDispatchToProps = (dispatch, ownProps) => ({
  addComment: (author, comment) =>
    dispatch(addComment(ownProps.photo.id, author, comment))
});

export default connect(mapStateToProps, mapDispatchToProps)(Comments);
