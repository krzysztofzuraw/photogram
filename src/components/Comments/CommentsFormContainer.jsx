import React from "react";
import { withRouter } from "react-router";
import { connect } from "react-redux";

import { addComment } from "../../actions";

class CommentsFormContainer extends React.Component {
  defaultState = { name: "", comment: "" };

  constructor(props) {
    super(props);
    this.state = this.defaultState;
  }

  handleChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  handleSubmit = event => {
    const { dispatch, match } = this.props;
    event.preventDefault();
    dispatch(
      addComment(match.params.photoId, this.state.name, this.state.comment)
    );
    this.setState(this.defaultState);
  };

  render() {
    return (
      <div className="form-wrapper">
        <form className="comment-form" onSubmit={this.handleSubmit}>
          <input
            onChange={this.handleChange}
            type="text"
            id="name"
            name="name"
            placeholder="Name"
            className="form-input"
            required
            value={this.state.name}
          />
          <textarea
            onChange={this.handleChange}
            id="comment"
            name="comment"
            placeholder="Message"
            className="form-input"
            required
            value={this.state.comment}
          />
          <input type="submit" value="Submit" className="form-button" />
        </form>
      </div>
    );
  }
}

export default withRouter(connect()(CommentsFormContainer));
