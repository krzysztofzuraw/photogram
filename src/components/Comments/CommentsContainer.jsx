import React from "react";
import { connect } from "react-redux";

import Comments from "./Comments.jsx";

// import { addComment } from "../../actions/index.js";

// class CommentsContainer extends React.Component {
//   // handleSubmit(event, { name, value }) {
//   //   event.preventDefault();
//   //   debugger;
//   //   this.props.addComment(this.refs.author.value, this.refs.comment.value);
//   //   this.refs.author.value = "";
//   //   this.refs.comment.value = "";
//   // }

//   render() {
//     return <Comments comments={this.props.comments} />;
//   }
// }

// // const mapDispatchToProps = (dispatch, ownProps) => ({
// //   addComment: (author, comment) =>
// //     dispatch(addComment(ownProps.photo.id, author, comment))
// // });

const mapStateToProps = (state, ownProps) => ({
  comments: state.comments
});
export default connect(mapStateToProps)(Comments);
