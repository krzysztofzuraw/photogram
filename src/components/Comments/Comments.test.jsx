import React from "react";

import { shallow } from "enzyme";

import Comments from "./Comments";

describe("Comments Component", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Comments comments={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
