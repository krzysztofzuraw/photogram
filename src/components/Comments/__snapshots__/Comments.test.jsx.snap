// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Comments
    comments={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="comments-header"
>
          <span
                    className="comments-header-title"
          >
                    Comments
          </span>
</div>,
        <div
          className="comments-list"
>
          
</div>,
        <withRouter(Connect(CommentsFormContainer)) />,
      ],
      "className": "comments",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            className="comments-header-title"
>
            Comments
</span>,
          "className": "comments-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Comments",
            "className": "comments-header-title",
          },
          "ref": null,
          "rendered": "Comments",
          "type": "span",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "comments-list",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="comments-header"
>
            <span
                        className="comments-header-title"
            >
                        Comments
            </span>
</div>,
          <div
            className="comments-list"
>
            
</div>,
          <withRouter(Connect(CommentsFormContainer)) />,
        ],
        "className": "comments",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="comments-header-title"
>
              Comments
</span>,
            "className": "comments-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Comments",
              "className": "comments-header-title",
            },
            "ref": null,
            "rendered": "Comments",
            "type": "span",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "comments-list",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
